### ISSUE TITLE
Tekken 5 Dark Resurrection Online (NPUB30009) - Game crashes when network status set to "Connected"

### ISSUE THREAD

**Issue by Ninetime** [RPCS3/rpcs3#13014]
### Quick summary

Tekken 5 DRO crashes when the network status is set to "Connected".

### Details

Tekken 5 DRO crashes when the network status is set to "Connected". 
![image](https://user-images.githubusercontent.com/6248582/205508483-4802ca32-d794-40ab-90a5-d3d7bb9fa23e.png)
![image](https://user-images.githubusercontent.com/6248582/205508561-4aea7f1d-d592-4dff-a493-072021b00b54.png)

If the network status is set to "Disconnected", the game is able to boot. When attempting to go into online mode, it successfully connects to RPCN, displaying a message on the overlay. Afterwards, it fails to connect to online and returns you to the menu.
![image](https://user-images.githubusercontent.com/6248582/205508524-24e8b0d5-2560-4f7c-ba86-fa66fee8b255.png)

### Attach a log file

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/10149184/RPCS3.log.gz)

### System configuration

RPCS3 v0.0.25-14407-82a8d96f Alpha | master
AMD Ryzen 7 5800X 8-Core Processor | 16 Threads | 31.93 GiB RAM | TSC: 3.800GHz | AVX+ | FMA3
Operating system: Windows, Major: 10, Minor: 0, Build: 19044, Service Pack: none, Compatibility mode: 0
Qt version: Compiled against Qt 5.15.2 | Run-time uses Qt 5.15.2
Current Time: 2022-12- 4T13:22:24

**Comment by capriots** [comment]
While the game is trying to parse a file it downloaded from http://f218update.tekken.bng.jp, it gets stuck in a loop doing this:
```c
while (*c != '.')
{
    c++;
}
```
On a PS3, argv is stored at the beginning of the stack, where the game eventually finds `'.'`:
<Details><Summary>Screenshot</Summary>

![ps3mem](https://user-images.githubusercontent.com/29807355/205941604-25c6257d-e6ed-4520-9266-b77eef84a45e.png)
</Details>

On RPCS3, there is nothing, so the game crashes instead:
<Details><Summary>Screenshot</Summary>

![rpcs3mem](https://user-images.githubusercontent.com/29807355/205937662-43f6e8d6-8cd5-47b8-8e3e-742457125ddb.png)
</Details>

**Comment by kd-11** [comment]
You're right, the stack frame is not properly set up due to HLE. In practice argv should be passed to 'main' that way.
But its still quite shocking that a game would rely on this behaviour.

@capriots Care to write a PR for this? You already have most of the work done.

**Comment by capriots** [comment]
I can try, but I'm not sure if I have enough experience for this. It would probably be better if someone more experienced handles it.
At least I managed to write a patch as a temporary workaround
```yaml
PPU-b8820496f8fd6f4b5e00798bde60b112daa7a606:
  "Online fix":
    Games:
      "TEKKEN 5 DARK RESURRECTION ONLINE (Updated)":
        NPUB30009: [ 01.00 ]
    Patch:
      - [ be32, 0x211024, 0x60000000 ]
      - [ be32, 0x2110ec, 0x60000000 ]

PPU-3b68a1ecc7531d1ae170fb00e2db03d17bffcd44:
  "Online fix":
    Games:
      "TEKKEN 5 DARK RESURRECTION ONLINE":
        NPUB30009: [ 01.00 ]
    Patch:
      - [ be32, 0x1fe724, 0x60000000 ]
      - [ be32, 0x1fe7ec, 0x60000000 ]
```

**Comment by Ninetime** [comment]
I tested the patch and it worked as outlined. Is the file that was downloaded the update referenced in the first patch name "TEKKEN 5 DARK RESURRECTION ONLINE (Updated)"

**Comment by capriots** [comment]
That's right. The game downloads a new eboot.bin.

**Comment by andymage593** [comment]
> PPU-b8820496f8fd6f4b5e00798bde60b112daa7a606:
>   "Online fix":
>     Games:
>       "TEKKEN 5 DARK RESURRECTION ONLINE (Updated)":
>         NPUB30009: [ 01.00 ]
>     Patch:
>       - [ be32, 0x211024, 0x60000000 ]
>       - [ be32, 0x2110ec, 0x60000000 ]
> 
> PPU-3b68a1ecc7531d1ae170fb00e2db03d17bffcd44:
>   "Online fix":
>     Games:
>       "TEKKEN 5 DARK RESURRECTION ONLINE":
>         NPUB30009: [ 01.00 ]
>     Patch:
>       - [ be32, 0x1fe724, 0x60000000 ]
>       - [ be32, 0x1fe7ec, 0x60000000 ]
How does this patch use it correctly?

**Comment by Ninetime** [comment]
@andymage593 This patch isn't needed anymore. The issue was fixed with PR #14461

**Comment by andymage593** [comment]
THANK YOU
